contracts/staking/tests/NativeTokenStakingManagerTests.t.sol

```diff
@@ -0,0 +1,324 @@
+        // TODO: implement
+    }
+    function testInitializeValidatorRegistrationInsufficientDuration() public {
+        // TODO: implement
+    }
+
+    // The following tests call functions that are  implemented in StakingManager, but access state that's
+    // only set in NativeTokenStakingManager. Therefore we call them via the concrete type, rather than a
+    // reference to the abstract type.
+    function testResendRegisterValidatorMessage() public {
+        bytes32 validationID = _setUpInitializeValidatorRegistration(
+            DEFAULT_NODE_ID, DEFAULT_SUBNET_ID, 1e6, DEFAULT_EXPIRY, DEFAULT_ED25519_SIGNATURE
+        );
+        (, bytes memory registerSubnetValidatorMessage) = StakingMessages
+            .packRegisterSubnetValidatorMessage(
+            StakingMessages.ValidationInfo({
+                subnetID: DEFAULT_SUBNET_ID,
+                nodeID: DEFAULT_NODE_ID,
+                weight: 1e6,
+                registrationExpiry: DEFAULT_EXPIRY,
+                signature: DEFAULT_ED25519_SIGNATURE
+            })
+        );
+        vm.mockCall(
+            WARP_PRECOMPILE_ADDRESS,
+            abi.encode(IWarpMessenger.sendWarpMessage.selector),
+            abi.encode(bytes32(0))
+        );
+        vm.expectCall(
+            WARP_PRECOMPILE_ADDRESS,
+            abi.encodeCall(IWarpMessenger.sendWarpMessage, registerSubnetValidatorMessage)
+        );
+        app.resendRegisterValidatorMessage(validationID);
+    }
+
+    function testCompleteValidatorRegistration() public {
+        _setUpCompleteValidatorRegistration({
+            nodeID: DEFAULT_NODE_ID,
+            subnetID: DEFAULT_SUBNET_ID,
+            weight: 1e6,
+            registrationExpiry: DEFAULT_EXPIRY,
+            signature: DEFAULT_ED25519_SIGNATURE,
+            registrationTimestamp: 1000
+        });
+    }
+
+    function testInitializeEndValidation() public {
+        _setUpInitializeEndValidation({
+            nodeID: DEFAULT_NODE_ID,
+            subnetID: DEFAULT_SUBNET_ID,
+            weight: 1e6,
+            registrationExpiry: DEFAULT_EXPIRY,
+            signature: DEFAULT_ED25519_SIGNATURE,
+            registrationTimestamp: 1000,
+            completionTimestamp: 2000
+        });
+    }
+
+    function testInitializeEndValidationWithUptimeProof() public {
+        // TODO: implement
+    }
+
+    function testInitializeEndValidationExcessiveChurn() public {
+        // TODO: implement
+    }
+
+    function testResendEndValidation() public {
+        bytes32 validationID = _setUpInitializeEndValidation({
+            nodeID: DEFAULT_NODE_ID,
+            subnetID: DEFAULT_SUBNET_ID,
+            weight: 1e6,
+            registrationExpiry: DEFAULT_EXPIRY,
+            signature: DEFAULT_ED25519_SIGNATURE,
+            registrationTimestamp: 1000,
+            completionTimestamp: 2000
+        });
+        bytes memory setValidatorWeightPayload =
```

Comment:
can add a `using` statement for the IDs

Reply:
Are you suggesting adding `using` for `validationID` so we can call certain packing methods directly on the `validationID`? My hesitation there is that not all of the methods take validationID's, and I'd prefer consistent notation.